src/gui/plugins/tape_measure/TapeMeasure.cc

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <ignition/msgs/marker.pb.h>
+
+#include <iostream>
+#include <ignition/common/Console.hh>
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/MainWindow.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport/Node.hh>
+#include <ignition/transport/Publisher.hh>
+
+#include "ignition/gazebo/gui/GuiEvents.hh"
+
+#include "TapeMeasure.hh"
+
+namespace ignition::gazebo
+{
+  class TapeMeasurePrivate
+  {
+    /// \brief Ignition communication node.
+    public: transport::Node node;
+
+    /// \brief True if currently measure, else false.
+    public: bool measure = false;
+
+    /// \brief The id of the start point marker.
+    public: int startPointId = 1;
```

Comment:
How about making these IDs const? We also try to use the `k` prefix for const variables.

Reply:
4d41ebe04525263d82069ea2d1d3f31c66711731